### PR TITLE
Tech :: move out step deifnitions in a specific module

### PR DESCRIPTION
**Pull request by adimascio**
It should be simple to read and maintain as this. On the way, fixed the steps documentation and introduce a more formal type definition for each step.

See individual commits for more explanation. (This PR is not supposed to be squashed before merging)

### CODE REVIEW

**Comment by piotch** [comment]
This is not directly linked to this PR but since we are fixing the doc: `This step is meant to select a specific domain (using MongoDB terminology).` sounds off, "domain" is not Mongo terminology, it's Toucan, maybe you meant "collection" instead of "domain" ?

File: src/lib/steps.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  oldname: string;
+  newname: string;
+}
+
+export interface DeleteStep {
+  name: 'delete';
+  columns: Array<string>;
+}
+
+export interface NewColumnStep {
```
Comment (piotch) [inline src/lib/steps.ts]: this one is not documented

Comment (adimascio) [reply src/lib/steps.ts]: :+1:

**Comment by adimascio** [comment]
> This is not directly linked to this PR but since we are fixing the doc: `This step is meant to select a specific domain (using MongoDB terminology).` sounds off, "domain" is not Mongo terminology, it's Toucan, maybe you meant "collection" instead of "domain" ?

I agree that "domain" might be a bad wording. "Collection" on the other hand might sound too much mongo-ish, don't you think? What about simply "dataset" ?

**Comment by piotch** [comment]
> I agree that "domain" might be a bad wording. "Collection" on the other hand might sound too much mongo-ish, don't you think? What about simply "dataset" ?

dataset 👍

**Comment by piotch** [comment]
sorry, button confusion

File: src/lib/steps.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/**
+ * This module defines the supported unit-of-transformation steps.
+ */
+
+export interface DomainStep {
```
Comment (Dalanir) [inline src/lib/steps.ts]: Rename to `selectDataset` I guess following previous comments

Comment (adimascio) [reply src/lib/steps.ts]: :+1: I will do this in a distinct PR once this one is merged.

**Pull request merged**